python/ray/train/collective/collectives.py

```diff
@@ -0,0 +1,111 @@
+import pickle
```

Comment:
nit: I think pickle should be fine simple/primitive data types, but in the future if we need to support more complex data types can use cloudpickle:

```python
import ray.cloudpickle as pickle
```